### PR TITLE
cache doctor rst

### PR DESCRIPTION
**Pull request by smoench**
This PR adapts the latest contribution https://github.com/OskarStark/doctor-rst/pull/626

### CODE REVIEW

**Review by OskarStark — APPROVED** [review]

**Comment by smoench** [comment]
Can `actions/cache` cache only directories and not a single file? Or what I'm doing wrong?

**Comment by smoench** [comment]
I had to create `.cache` directory to get caching work

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -52,7 +52,16 @@ jobs:
...
                 uses: actions/checkout@v2
 
+            -   name: "Create cache dir"
+                run: mkdir .cache
+
+            -   name: "Cache DOCtor-RST"
+                uses: actions/cache@v1
+                with:
+                    path: .cache
+                    key: ${{ runner.os }}-doctor-rst-2
```
Comment (smoench) [inline .github/workflows/ci.yaml]: Should I change the suffix to the base branch?

**Comment by smoench** [comment]
Should I also provide PRs for 4.4 and master?

**Comment by javiereguiluz** [comment]
@smoench no, don't worry. We merge each PR in all active merges up to `master`. Thanks.

**Comment by smoench** [comment]
@javiereguiluz I was just asking as the suffix of cache key might has to change to base branch name

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -52,7 +52,16 @@ jobs:
...
                 uses: actions/checkout@v2
 
+            -   name: "Create cache dir"
+                run: mkdir .cache
+
+            -   name: "Cache DOCtor-RST"
+                uses: actions/cache@v1
+                with:
+                    path: .cache
+                    key: ${{ runner.os }}-doctor-rst-34
```
Comment (OskarStark) [inline .github/workflows/ci.yaml]: I would prefer using the base branch programmatically. IIRC GithubActions exposes an env var

Comment (smoench) [reply .github/workflows/ci.yaml]: I used `${{ github.base_ref || github.ref }}`.  `github.base_ref` is only available on pull requests. With `github.ref` I'm not 100% sure if it's contains only the branch name.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -52,7 +52,16 @@ jobs:
...
                 uses: actions/checkout@v2
 
+            -   name: "Create cache dir"
+                run: mkdir .cache
+
+            -   name: "Cache DOCtor-RST"
+                uses: actions/cache@v1
+                with:
+                    path: .cache
+                    key: ${{ runner.os }}-doctor-rst-${{ github.base_ref || github.ref }}
```
Comment (OskarStark) [inline .github/workflows/ci.yaml]: Excellent 🙃

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Wow
Screenshot 2020-02-27 12 05 56

**Comment by smoench** [comment]
Please don't merge yet as the base branch resolving doesn't work

**Comment by smoench** [comment]
Got it work. Ready for merge

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -52,7 +52,20 @@ jobs:
...
+
+            -   name: "Extract base branch name"
+                run: echo "##[set-output name=branch;]$(echo ${GITHUB_BASE_REF:=${GITHUB_REF##*/}})"
+                id: extract_base_branch
+
+            -   name: "Cache DOCtor-RST"
+                uses: actions/cache@v1
+                with:
+                    path: .cache
+                    key: ${{ runner.os }}-doctor-rst-${{ steps.extract_base_branch.outputs.branch }}
```
Comment (OskarStark) [inline .github/workflows/ci.yaml]: Nice trick 👍

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Thank you Simon.

**Comment by smoench** [comment]
From about ~10min to seconds 🚀

**Pull request merged**